src/request.ts

```diff
@@ -744,27 +745,94 @@ class DatastoreRequest {
 
       if (options.consistency) {
         const code = CONSISTENCY_PROTO_CODE[options.consistency.toLowerCase()];
-        reqOpts.readOptions = {
+        sharedQueryOpts.readOptions = {
           readConsistency: code,
         };
       }
 
       if (query.namespace) {
-        reqOpts.partitionId = {
+        sharedQueryOpts.partitionId = {
           namespaceId: query.namespace,
         };
       }
 
-      this.request_(
-        {
-          client: 'DatastoreClient',
-          method: 'runQuery',
-          reqOpts,
-          gaxOpts: options.gaxOptions,
-        },
-        onResultSet
-      );
+      if (query.aggregations.length === 0) {
+        const reqOpts: RequestOptions = sharedQueryOpts;
+        reqOpts.query = queryProto;
+        this.request_(
+            {
+              client: 'DatastoreClient',
+              method: 'runQuery',
+              reqOpts,
+              gaxOpts: options.gaxOptions,
+            },
+            onResultSet
+        );
+      } else {
+        const aggregationQueryOptions: AggregationQueryOptions = {
```

Comment:
why not separate this out into a different method? `runAggregationQueryStream` vs `runQueryStream`?

Reply:
I see what you mean. This function is getting gnarly.

So currently this code is being run in `runQueryStream`, if we keep `runQueryStream` the way it was before and add a new function `runAggregationQueryStream` then there are a few downsides:
1. We expand the API surface by adding a new function
2. With a new function the user has to know to pass an aggregate query into `runAggregationQueryStream` and a regular query into `runQueryStream` if they are using requests directly. Note that both aggregate and non-aggregate queries are both instances of the `Query` class so there could be situations where it is not clear which type of query we are dealing with.

Although, maybe what you were getting at here was to move the code in the `else` branch out into a separate, private function (ie. `runAggregationQueryStream`) in a utility function which I think is a good idea. It might be a good idea with the `if` code as well.

Reply:
I think what we could do here is add two inner functions, one for the `if` and one for the `else` so that the if/else block is concise. Do you think that would address the concern in question?

Reply:
See comment above